### ISSUE TITLE
Don't change files in-place

### ISSUE THREAD

**Issue by mosheavni** [nvuillam/npm-groovy-lint#231]
I'm trying to integrate npm-groovy-lint with my nvim to format my files,
Can it be optional to not change the files in-place but just echo the formatted file as many formatter tools do?
Thanks.

**Comment by nvuillam** [comment]
Do you mean return in stdout the updated sources ?

**Comment by mosheavni** [comment]
> Do you mean return in stdout the updated sources ?

correct.

**Comment by nvuillam** [comment]
Mmmm I could add an option for that, but how would you use it then ?
Maybe it's simpler for you if I add an output property to the json output ?

**Comment by mosheavni** [comment]
Not sure I understand your suggestion.
Look here:
https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/HELPERS.md#to_temp_file

when null-ls formats the file, it passes is as stdin for the format command like so:
```bash
cat Jenkinsfile | npm-groovy-lint --format -
```
and the file is replaced inside the editor with the stdout of `npm-groovy-lint`.

They created a function for cases like this where you can only format existing files and not `stdin`, they create a temp file with the same content of the file, format it, and copies it back to the original file.
This "hack" is fine, but a better idea is to have the formatter optionally just echo back the formatted file.
I hope that makes sense.

**Comment by nvuillam** [comment]
Understood, I'll see what I can do :)

**Comment by nvuillam** [comment]
@mosheavni I suggest you try again `cat Jenkinsfile | npm-groovy-lint --format -` with v10.1.0 :)

**Comment by mosheavni** [comment]
> @mosheavni I suggest you try again `cat Jenkinsfile | npm-groovy-lint --format -` with v10.1.0 :)

You're awesome. works perfectly.
Look!!!
https://asciinema.org/a/K6MJYsMlCrNPaVsC3K7v5cmCQ

**Comment by nvuillam** [comment]
Haha nice , glad to see you made a good use of the feature 🤣
